core/templates/components/summary-tile/lesson-card.component.spec.ts

```diff
@@ -0,0 +1,334 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for LessonCardComponent
+ */
+
+import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {FormsModule} from '@angular/forms';
+import {waitForAsync, ComponentFixture, TestBed} from '@angular/core/testing';
+import {UrlInterpolationService} from 'domain/utilities/url-interpolation.service';
+import {AssetsBackendApiService} from 'services/assets-backend-api.service';
+import {MockTranslatePipe} from 'tests/unit-test-utils';
+import {LessonCardComponent} from './lesson-card.component';
+import {NO_ERRORS_SCHEMA} from '@angular/core';
+import {AppConstants} from 'app.constants';
+
+import {CollectionSummary} from 'domain/collection/collection-summary.model';
+import {LearnerExplorationSummary} from 'domain/summary/learner-exploration-summary.model';
+import {StorySummary} from 'domain/story/story-summary.model';
+
+describe('LessonCardComponent', () => {
+  let component: LessonCardComponent;
+  let fixture: ComponentFixture<LessonCardComponent>;
+
+  let urlInterpolationService: UrlInterpolationService;
+  let assetsBackendApiService: AssetsBackendApiService;
+
+  const sampleCollection = {
+    last_updated_msec: 1591296737470.528,
+    community_owned: false,
+    objective: 'Test Objective',
+    id: '44LKoKLlIbGe',
+    thumbnail_icon_url: '/subjects/Algebra.svg',
+    language_code: 'en',
+    thumbnail_bg_color: '#cc4b00',
+    created_on: 1591296635736.666,
+    status: 'public',
+    category: 'Algebra',
+    title: 'Test Title',
+    node_count: 0,
+  };
+
+  const sampleExploration = {
+    last_updated_msec: 1591296737470.528,
+    community_owned: false,
+    objective: 'Test Objective',
+    id: '44LKoKLlIbGe',
+    num_views: 0,
+    thumbnail_icon_url: '/subjects/Algebra.svg',
+    human_readable_contributors_summary: {},
+    language_code: 'en',
+    thumbnail_bg_color: '#cc4b00',
+    created_on_msec: 1591296635736.666,
+    ratings: {
+      1: 0,
+      2: 0,
+      3: 0,
+      4: 0,
+      5: 0,
+    },
+    status: 'public',
+    tags: [],
+    activity_type: 'exploration',
+    category: 'Algebra',
+    title: 'Test Title',
+  };
+
+  const sampleNode = {
+    id: 'node_1',
+    thumbnail_filename: 'image.png',
+    title: 'Chapter 1',
+    description: 'Description 1',
+    prerequisite_skill_ids: ['skill_1'],
+    acquired_skill_ids: ['skill_2'],
+    destination_node_ids: ['node_2'],
+    outline: 'Outline',
+    exploration_id: null,
+    outline_is_finalized: false,
+    thumbnail_bg_color: '#a33f40',
+    status: 'Published',
+    planned_publication_date_msecs: 100,
+    last_modified_msecs: 100,
+    first_publication_date_msecs: 200,
+    unpublishing_reason: null,
+  };
+
+  const sampleTopic = {
```

Comment:
So you have sampleTopic, sampleTopic2 and sampleTopic3. Looks a bit strange, doesn't it?

Reply:
Done. Renamed

Reply:
Very good, I actually implied just adding sampleTopic1, but that's even better